modules/libcom/src/iocsh/iocsh.cpp

```diff
@@ -545,6 +546,16 @@ int iocshSetError(int err)
     return err;
 }
 
+void addPwdMacro(iocshContext *context) {
+    if (context != NULL) {
+        char buf[256];
```

Comment:
256 may be too small, but sys/param.h isn't a portable header; can you make this 1024 instead? That should be enough for most sites. It should probably inform the user if it gets an ERANGE error though.

Reply:
@anjohnson FYI #102 would add `epicsPathAllocCWD()` to avoid this sort of situation.

Reply:
@anjohnson and @mdavidsaver, I'm going through macLib and I've found this in macLib.h:

`/** \brief Maximum size of a macro name or value
 */
#define MAC_SIZE 256`

Although it seems that you can register a macro with any amount of characters, this constant is used for the macro expansion.

Reply:
Ah, yes.  I had forgotten about that limit.  And it isn't so straightforward to eliminate, or even extend as it is used to size some local array variables (on the stack).

I guess use of macLib needs to be reconsidered...

Reply:
Are you thinking about hard-code ${PWD} or $(PWD) as a "special command" inside iocshBody instead of using macLib?

Reply:
I've thinking about having `cd` (`chdirCallFunc()` in libComRegister.c) call `epicsSetEnv()`, or alternately `putenv()`.  Not as good of a solution as it wouldn't capture direct calls to `chdir()`.

Reply:
#137 would also come into play.

The macLib limit also effects my #102.

Reply:
`${PWD}` isn't the only path I'd like to expose via macLib, so I'm not inclined to make a special case of it.

Reply:
I've just pushed the code with the epicsSetEnv idea. It works regarding that the user issues a cd on the shell. Can you think about a boot-up trigger to make sure PWD will always exist?